packages/contracts/contracts/registry/InterfaceBasedRegistry.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.10;
+
+import "@openzeppelin/contracts/proxy/utils/UUPSUpgradeable.sol";
+import "../core/component/Permissions.sol";
+import "../core/erc165/AdaptiveERC165.sol";
+
+/// @title An ERC165-based registry for contracts
+/// @author Michel Heuer, Sarkawt Noori - Aragon Association - 2022
+/// @notice This contract allows to register contracts
+abstract contract InterfaceBasedRegistry is Permissions, UUPSUpgradeable {
```

Comment:
Not sure how to proceed here.
any recommendation how to merge this file with ERC165RegistryBase.sol ?
while they are similar but the way they use component and permission they are different.
WDYT?

Reply:
I would personally remove the unused code by now and only add it when effectively used. Also I'd recommend using the very least base functionality, following the same principle.

Reply:
Also, regarding tests, let's make sure that they cover all the newly added code. I'll re-review later on

Reply:
@Rekard0 Could the unused contract be moved to the testing section, to act as a simple instance of ERC165RegistryBase in order to test it?

Reply:
we have a mock contract that is used for testing the base

Reply:
Ok, as long as the new tests from Michael keep testing that, I'm OK 👍